prometheus/metrics.go

```diff
@@ -249,3 +247,8 @@ func fetchScalarDouble(api v1.API, query string, now time.Time) (float64, error)
 	}
 	return 0, fmt.Errorf("Invalid query, vector expected: %s", query)
 }
+
+func replaceInvalidCharacters(metricName string) string {
```

Comment:
FWIW here is a similar function from Prometheus:
https://github.com/prometheus/prometheus/blob/master/util/strutil/strconv.go#L42-L44
The list of allowed characters is aligned with the official doc:
https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels

Reply:
thanks I will update to match it